mongo/errors.go

```diff
@@ -228,8 +234,6 @@ type ServerError interface {
 	HasErrorMessage(string) bool
 	// HasErrorCodeWithMessage returns true if any of the contained errors have the specified code and message.
 	HasErrorCodeWithMessage(int, string) bool
-
-	serverError()
```

Comment:
I think it should be fine to remove the unexposed function form the interface because it is never implemented or used.

Reply:
The point of the `serverError` function in this interface is to "seal" the interface so that types outside of the `mongo` package can't implement it. It's not clear that keeping the interface sealed is necessary, but this has come up before and we decided to keep it sealed, so we should try to understand the history of those decisions.